File: app/src/main/java/com/github/nein37/styleviewer/MainActivity.java
package com.github.nein37.styleviewer;

import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;

import com.github.nein37.styleviewer.databinding.ActivityMainBinding;

import java.lang.reflect.Field;
import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        setSupportActionBar(binding.toolbar);

        final List<Field> textAppearanceList = new ArrayList<>();

        for (Field field : R.style.class.getFields()) {
            String fieldName = field.getName();
            if (fieldName.startsWith("TextAppearance_")) {
                textAppearanceList.add(field);
            }
        }

        binding.recyclerView.setLayoutManager(new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false));
        binding.recyclerView.setAdapter(new TextAppearanceAdapter(textAppearanceList));
    }
}

File: app/src/main/java/com/github/nein37/styleviewer/MainActivity.kt
package com.github.nein37.styleviewer

import android.content.Intent
import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.RecyclerView
import com.github.nein37.styleviewer.databinding.ActivityMainBinding
import com.github.nein37.styleviewer.presenter.TextAppearancePresenter
import com.github.nein37.styleviewer.usecase.TextAppearanceUseCase
import com.github.nein37.styleviewer.view.MainView


class MainActivity : AppCompatActivity(), MainView {

    lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = DataBindingUtil.setContentView<ActivityMainBinding>(this, R.layout.activity_main)
        setSupportActionBar(binding.toolbar)

        val presenter = TextAppearancePresenter(this, this, TextAppearanceUseCase(this))
        presenter.onCreate()
    }

    override fun <T : RecyclerView.ViewHolder> setupRecyclerView(layoutManager: RecyclerView.LayoutManager, adapter: RecyclerView.Adapter<T>) {
        binding.recyclerView.layoutManager = layoutManager
        binding.recyclerView.adapter = adapter
    }
}
